packages/manager/apps/sap-features-hub/src/utils/importJson.ts

```diff
@@ -0,0 +1,35 @@
+import { installationInitialValues } from '@/context/installationInitialValues.constants';
+import { formMappers } from '../mappers/formMappers';
+
+export const readJsonFile = async (file: File) => {
+  return new Promise((resolve, reject) => {
+    const reader = new FileReader();
+
+    reader.onload = () => {
+      const text = reader.result;
+      if (typeof text !== 'string') {
+        reject(new Error('Invalid type of file'));
+        return;
+      }
```

Comment:
The use of return is a bit confusing as it's an async function
```suggestion
      if (typeof text !== 'string') {
        reject(new Error('Invalid type of file'));
      } else {
```

Reply:
Yes, agreed, I removed the return and rewrote the condition